### PR TITLE
added AuthOverride example

### PR DESCRIPTION
**Pull request by tegk**
Added example how to use the ServiceAuthFuncOverride function

### CODE REVIEW

File: auth/examples_test.go [hunk 1]
```diff
@@ -2,11 +2,14 @@ package grpc_auth_test
 
 import (
 	"context"
+	"log"
+
+	grpc_auth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	grpc_ctxtags "github.com/grpc-ecosystem/go-grpc-middleware/tags"
```
Comment (johanbrandhorst) [inline auth/examples_test.go:8]: Can you put this import group underneath the third party imports, or together. It doesn't matter too much (we've fixed it in v2).

File: auth/examples_test.go [hunk 2]
```diff
@@ -18,24 +21,63 @@ func userClaimFromToken(struct{}) string {
...
-			return nil, err
-		}
-		tokenInfo, err := parseToken(token)
-		if err != nil {
-			return nil, status.Errorf(codes.Unauthenticated, "invalid auth token: %v", err)
-		}
-		grpc_ctxtags.Extract(ctx).Set("auth.sub", userClaimFromToken(tokenInfo))
-		newCtx := context.WithValue(ctx, "tokenInfo", tokenInfo)
-		return newCtx, nil
+//exampleAuthFunc is used by a middleware to authenticate requests
```
Comment (johanbrandhorst) [inline auth/examples_test.go:24]: Nitpick: can you add a space after the slashes please?

File: auth/examples_test.go [hunk 3]
```diff
@@ -18,24 +21,63 @@ func userClaimFromToken(struct{}) string {
...
+func Example_serverConfigWithAuthOverride() {
+	server := grpc.NewServer(
+		grpc.StreamInterceptor(grpc_auth.StreamServerInterceptor(exampleAuthFunc)),
+		grpc.UnaryInterceptor(grpc_auth.UnaryServerInterceptor(exampleAuthFunc)),
+	)
+
+	//in this case a single service is used,
+	//in most use cased there would be at least two services defined in the proto file
+	pb.RegisterGreeterServer(server, &gRPCserverAuthenticated{})
+	pb.RegisterGreeterServer(server, &gRPCserverUnauthenticated{})
```
Comment (johanbrandhorst) [inline auth/examples_test.go:82]: Registering both doesn't make sense, so how about we have a boolean or something to show the toggle between authenticating with and without the override?
```suggestion
	if useCustomAuth {
		pb.RegisterGreeterServer(server, &gRPCserverUnauthenticated{})
	else {
		pb.RegisterGreeterServer(server, &gRPCserverAuthenticated{})
	}
```

Comment (tegk) [reply auth/examples_test.go:82]: true

**Comment by johanbrandhorst** [comment]
Thanks for your contribution! Could you cherry-pick this against v2 as well please?

**Comment by tegk** [comment]
> Thanks for your contribution! Could you cherry-pick this against v2 as well please?

https://github.com/grpc-ecosystem/go-grpc-middleware/pull/294

**Pull request merged**